mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,7 +114,11 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val bucketizer = new Bucketizer(uid).setSplits(splits)
+    val cutpoints = splits.distinct
+    if(splits.length != cutpoints.length) {
```

Comment:
Nit: space after "if". Also this could state the number of resulting buckets, since it definitely will happen:

`log.warn(s"Some quantiles were identical. Bucketing to ${splits.length - 1} buckets as a result.")`

Finally, please update the title of the PR to match the JIRA, which is more descriptive as a commit log message.